os/drivers/audio/syu645b.c

```diff
@@ -1031,19 +1040,201 @@ static void syu645b_hw_reset_config(FAR struct syu645b_dev_s *priv)
+		meddbg("Failed to read from equalizer json.\n");
+		close(fd);
+		free(buffer);
+		return -1;
+	}
+	
+	gJSON = cJSON_Parse(buffer);
+	
+	if (!gJSON) {
+		meddbg("Failed to parse equalizer json file.\n");
+		close(fd);
+		free(buffer);
+		return -1;
+	}
+	close(fd);
+	free(buffer);
+	return 0;
+}
+
+static int dec_to_hex_string(uint32_t decimal, char *hexadecimal){
+	if (decimal > 0xffffffff) {
+		meddbg("Input invalid preset.\n");
+		return -1;
+	}
+	
+	int pos = 9;
+	hexadecimal[0] = '0';
+	hexadecimal[1] = 'x';
+	hexadecimal[10] = '\0';
+	while (1) {
+		int mod = decimal % 16;
+		if (mod < 10) {
+			hexadecimal[pos] = 48 + mod;
+		}
+		else {
+			hexadecimal[pos] = 65 + (mod - 10);
+		}
+		decimal = decimal / 16;
+		pos--;
+
+		if (decimal == 0) {
+			if (pos > 1) {
+				for(int i=2; i<=pos; i++){
+					hexadecimal[pos] = '0';
+				}
+			}
+			break;
+		}
+	}
+
+	return 0;
+}
+
+static int string_to_script(char *string, t_codec_init_script_entry *script){
+	int cnt = 0;
+
+	const char *ptr = string;
+	while (*ptr && cnt < SYU645B_REG_DATA_TYPE_MAX) {
+		if (*ptr == '0' && (*(ptr + 1) == 'x' || *(ptr + 1) == 'X')) {
+			char *end_ptr;
+			uint8_t val = strtol(ptr, &end_ptr, 16);
+
+			if (ptr != end_ptr) {
+				if (cnt == 0) {
+					script->addr = val;
+				}
+				else {
+					script->val[cnt-1] = val;	
+				}
+				ptr = end_ptr;
+				cnt++;
+				continue;
+			}
+		}
+		ptr++;
+	}
+
+	script->delay = 0;
+	script->type = cnt;
+
+	return 0;
+}
+
 static void syu645b_set_equalizer(FAR struct syu645b_dev_s *priv, uint32_t preset)
 {
 	/* Reset First */
 	syu645b_reset_config(priv);
 
-	/* And then Set default EQ Set */
+	/* Initialize for i2c write */
+	uint8_t reg[SYU645B_REG_DATA_TYPE_MAX];
+	FAR struct i2c_dev_s *dev = priv->i2c;
+	FAR struct i2c_config_s *syu645b_i2c_config = &(priv->lower->i2c_config);
+	
+	int ret = 0;
+
+	/* Parse equlizer json to gJSON */
+	ret = parse_equalizer_metadata_json();
+	if (ret) {
+		printf("[TEST] goto default_script parse_equalizer_metadata_json\n");
+		goto default_script;
+	}
+
+	/* Convert preset hex to string for key */
+	char c_preset[11];
+	ret = dec_to_hex_string(preset, c_preset);
+	if (ret) {
+		goto default_script;
+	}
+
+	/* Executes given script through i2c to configure SYU645B device. */
+	cJSON *search = cJSON_GetObjectItemCaseSensitive(gJSON, c_preset);
+
+	/* Find proper key */
+	if (search){
+		cJSON *key;
+		t_codec_init_script_entry script;
+		
+		cJSON_ArrayForEach(key, search){
+			if (strcmp(key->string, "_comment") == 0) {
+				printf("[TEST] find _comment\n");
+				continue;
+			}
+			printf("[TEST] script | %s : %s\n", key->string, key->valuestring);
+
+			ret = string_to_script(key->valuestring, &script);
+			if (ret) {
+				goto default_script;
+			}
+
+			if (script.addr == 0x08 && script.type == SYU645B_REG_D_2BYTE) {
+				priv->max_volume = script.val[0];
+				printf("[TEST] priv->max_volume update %d \n", priv->max_volume);
+			}
+
+			reg[0] = script.addr;
```

Comment:
I think you can remove addr, each of script line has address value in offset 0 of array.
also I think you can update syu645b_scrhipt.h